### PR TITLE
CI: upgrade to Playwright 1.18

### PR DESCRIPTION
**Pull request by worldomonation**
#### Changes proposed in this Pull Request

Playwright 1.18 and 1.17 brings forth multiple improvements for our framework:
- (1.18) better TypeScript support to import ESM modules without compilation.
- (1.18) updated browser versions - Chrome 99
- (1.17) frame locators - ability to select elements within an iframe without having to first extract the iframe.

This PR updates the version of Playwright used for everything **except** for the desktop tests to Playwright 1.18.

#### Testing instructions

- [ ] Ensure that:
  - [ ] all existing build configurations that pass, should continue to pass.
  - [x] i18n and Quarantined E2E should execute, but failure is expected.

Related to #

### CODE REVIEW

**Comment by worldomonation** [comment]
Looks like the desktop tests fail when upgraded to 1.18.

I haven't looked deeply into why this is yet, will do so later today (UTC-8). Tagging @nsakaimbo for awareness.

**Comment by worldomonation** [comment]
Looks like on Playwright 1.17 and 1.18, ~~the `afterAll()` hook [here](https://github.com/Automattic/wp-calypso/blob/trunk/desktop/test/e2e/specs/login.js#L111-L125) refuse to run as expected, which is why we're seeing the failure.~~

edit: looks like something causes the electron app to hang after the login step, so it's not even related to the `afterAll()` hook.

**Comment by worldomonation** [comment]
I'm not sure if I've found the culprit, but something unusual seems to be occurring.

The following logs are emitted when running the desktop test locally with `DEBUG=pw:browser` environment:
```
 pw:browser [pid=20469][out] [2022-02-07 15:16:06.088] [desktop:session] [info] Got 'wordpress_logged_in' cookie, emitting 'logged-in' event... +2ms
  pw:browser [pid=20469][out] [2022-02-07 15:16:06.088] [desktop:menu] [info] Enabling logged in menu items +0ms
  pw:browser [pid=20469][out] [2022-02-07 15:16:06.383] [desktop:notifications:api] [info] Websocket error:  Unexpected server response: 403{} +294ms
  pw:browser [pid=20469][out] Error: Unexpected server response: 403 +1ms
  pw:browser [pid=20469][out]     at ClientRequest.<anonymous> (/Users/edwintakahashi/workspace/wp-calypso-second/desktop/release/mac-arm64/WordPress.com.app/Contents/Resources/app/node_modules/ws/lib/websocket.js:716:7) +0ms
  pw:browser [pid=20469][out]     at ClientRequest.emit (events.js:315:20) +0ms
  pw:browser [pid=20469][out]     at HTTPParser.parserOnIncomingClient [as onIncoming] (_http_client.js:641:27) +0ms
  pw:browser [pid=20469][out]     at HTTPParser.parserOnHeadersComplete (_http_common.js:126:17) +0ms
  pw:browser [pid=20469][out]     at TLSSocket.socketOnData (_http_client.js:509:22) +0ms
  pw:browser [pid=20469][out]     at TLSSocket.emit (events.js:315:20) +0ms
  pw:browser [pid=20469][out]     at addChunk (internal/streams/readable.js:309:12) +0ms
  pw:browser [pid=20469][out]     at readableAddChunk (internal/streams/readable.js:284:9) +0ms
  pw:browser [pid=20469][out]     at TLSSocket.Readable.push (internal/streams/readable.js:223:10) +0ms
  pw:browser [pid=20469][out]     at TLSWrap.onStreamRead (internal/stream_base_commons.js:188:23) +0ms
```

The same error is not observed when running with Playwright 1.16.

@nsakaimbo is there something that's pinning or expecting a certain version of Chrome/Playwright to be used?

**Comment by worldomonation** [comment]
Did some further investigation by cutting back the test script to the bare necessities, where the test step itself only has one call: `console.log('hi!')`.

Interestingly, the WordPress 7.2.0 app + Playwright 1.18 combination results in the following errors after the test steps complete:

```
  console.log
    hi!

      at Object.<anonymous> (test/e2e/specs/login.js:94:11)

  pw:browser [pid=28692][out] [2022-02-07 15:59:28.745] [platform:mac] [info] Application quit triggered +330ms
  pw:browser [pid=28692][out] [2022-02-07 15:59:28.746] [desktop:runapp] [info] Closing main window, last location: 'https://wordpress.com/' +1ms
  pw:browser [pid=28692][out] [2022-02-07 15:59:28.747] [desktop:runapp] [info] Window closed +1ms
  pw:browser [pid=28692][err] Waiting for the debugger to disconnect... +13ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.975 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +215ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.975 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Set a breakpoint on CTFontLogSystemFontNameRequest to debug. +1ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.976 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +0ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.976 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +1ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.978 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraginoSansGBInterface-W6", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +1ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.979 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +1ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.979 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +0ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.979 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +1ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.979 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +0ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.979 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".HiraKakuInterface-W7", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +0ms
  pw:browser [pid=28692][err] 2022-02-07 15:59:28.983 WordPress.com Helper (Renderer)[28697:6052315] CoreText note: Client requested name ".AppleSDGothicNeoI-Bold", it will get Times-Roman rather than the intended font. All system UI font access should be through proper APIs such as CTFontCreateUIFontForLanguage() or +[NSFont systemFontOfSize:]. +3ms
 FAIL  test/e2e/specs/login.js (7.616 s)


  ● Test suite failed to run

    thrown: "Exceeded timeout of 5000 ms for a hook.
    Use jest.setTimeout(newTimeout) to increase the timeout value, if this is a long-running test."

      125 | 	} );
      126 |
    > 127 | 	afterAll( async () => {
          | 	^
      128 | 		// if ( consoleStream ) {
      129 | 		// 	consoleStream.end();
      130 | 		// }

      at test/e2e/specs/login.js:127:2
      at Object.<anonymous> (test/e2e/specs/login.js:30:1)

Test Suites: 1 failed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        7.708 s
Ran all test suites.

```

This is not observed when running under WordPress App 7.2.0 + Playwright 1.16:

```
  console.log
    hi!

      at Object.<anonymous> (test/e2e/specs/login.js:94:11)

  pw:browser [pid=28574][out] [2022-02-07 15:59:08.848] [desktop:runapp] [info] Window closed +416ms
  pw:browser [pid=28574][out] [2022-02-07 15:59:08.848] [platform:mac] [info] All windows closed, shutting down +0ms
  pw:browser [pid=28574][out] [2022-02-07 15:59:08.849] [platform:mac] [info] Application quit triggered +0ms
  pw:browser [pid=28574] <gracefully close start> +13ms
  pw:browser [pid=28574][err] Waiting for the debugger to disconnect... +5ms
  pw:browser [pid=28574] <kill> +9ms
  pw:browser [pid=28574] <will force kill> +2ms
  pw:browser [pid=28574] starting temporary directories cleanup +0ms
  pw:browser [pid=28574] <process did exit: exitCode=null, signal=SIGKILL> +26ms
  pw:browser [pid=28574] starting temporary directories cleanup +1ms
  pw:browser <ws disconnected> ws://127.0.0.1:59372/dafaf331-c9f9-4a6d-8d13-b08462b57ac8 +2ms
  pw:browser <ws disconnected> ws://127.0.0.1:59374/devtools/browser/c2a86a51-d8fe-4bd0-b32c-22365ec7ab83 +1ms
  pw:browser [pid=28574] finished temporary directories cleanup +2ms
  pw:browser [pid=28574] finished temporary directories cleanup +1ms
  pw:browser [pid=28574] <gracefully close end> +0ms
 PASS  test/e2e/specs/login.js
  User Can log in
    ✓ Log in (19 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        2.616 s, estimated 6 s
Ran all test suites.
```

----

It seems that something related to fonts in the WordPress.com App 7.2.0 app is causing this error to occur, which prevents the debugger from disconnecting, which in turns makes Jest think that it hasn't finished its cleanup, resulting in the error at the `afterAll` hook.

**Comment by worldomonation** [comment]
I've filed an issue with the Playwright maintainers, in case the issue is on their end. https://github.com/microsoft/playwright/issues/11920

**Comment by nsakaimbo** [comment]
Thanks for digging into this, @worldomonation - this is a really weird one, indeed. I clarified the Electron version in the issue you filed in the PlayWright repository. A couple of quick thoughts:

- If it indeed is a font issue, I wonder if we can identify which font(s) exactly are the culprit?
- We could attempt to upgrade the Electron version for the app in `desktop/package.json` to the latest stable release - [Electron 17.0.0](https://www.electronjs.org/releases/stable#17.0.0) - perhaps that might help to get things working.

**Comment by worldomonation** [comment]
> Thanks for digging into this, @worldomonation - this is a really weird one, indeed. I clarified the Electron version in the issue you filed in the PlayWright repository. A couple of quick thoughts:
> 
>     * If it indeed is a font issue, I wonder if we can identify which font(s) exactly are the culprit?

I think the fonts listed in [this comment](https://github.com/Automattic/wp-calypso/pull/60902#issuecomment-1032067895) may be the culprit, at least on my macOS system. I wonder if Linux would be different.


>     * We could attempt to upgrade the Electron version for the app in `desktop/package.json` to the latest stable release - [Electron 17.0.0](https://www.electronjs.org/releases/stable#17.0.0) - perhaps that might help to get things working.

I'll try that, thanks.

**Comment by worldomonation** [comment]
@nsakaimbo You were correct - it was Electron 12.1 that was the problem! 

Upgraded to 17.0.0 and it finishes the run as expected:
```
  console.log
    hi!

      at Object.<anonymous> (test/e2e/specs/login.js:23:11)

  pw:browser [pid=58650][out] [2022-02-08 10:46:46.830] [platform:mac] [info] Application quit triggered +1s
  pw:browser [pid=58650][out] [2022-02-08 10:46:46.831] [desktop:runapp] [info] Closing main window, last location: 'https://wordpress.com/' +1ms
  pw:browser [pid=58650][out] [2022-02-08 10:46:46.834] [desktop:runapp] [info] Window closed +3ms
  pw:browser <ws disconnecting> ws://127.0.0.1:55689/076f4865-cc45-4fd1-82b3-c8bd4376a0c8 +5ms
  pw:browser [pid=58650][err] Debugger listening on ws://127.0.0.1:55689/076f4865-cc45-4fd1-82b3-c8bd4376a0c8 +1ms
  pw:browser [pid=58650][err] For help, see: https://nodejs.org/en/docs/inspector +1ms
  pw:browser <ws disconnected> ws://127.0.0.1:55689/076f4865-cc45-4fd1-82b3-c8bd4376a0c8 +6ms
  pw:browser [pid=58650] <process did exit: exitCode=null, signal=SIGSEGV> +255ms
  pw:browser [pid=58650] starting temporary directories cleanup +1ms
  pw:browser <ws disconnected> ws://127.0.0.1:55691/devtools/browser/e4338d25-89d2-443e-bdd2-ac92986b4604 +2ms
  pw:browser [pid=58650] finished temporary directories cleanup +7ms
 PASS  test/e2e/specs/login.js
  User Can log in
    ✓ Test Step (19 ms)

Test Suites: 1 passed, 1 total
Tests:       1 passed, 1 total
Snapshots:   0 total
Time:        3.469 s, estimated 9 s
Ran all test suites.
```

Given this information, would you/desktop team be open to upgrading Electron to this version?

**Comment by nsakaimbo** [comment]
> Upgraded to 17.0.0 and it finishes the run as expected

Glad to hear!

Could you please create an issue to review the breaking changes that might be caused as a result of this upgrade? In the past, we've been very intentional about upgrading Electron as silent breaking changes have been introduced when upgrading major versions.

Breaking changes are documented in the [Electron docs](https://www.electronjs.org/docs/latest/breaking-changes), and we'll want to review the changes between major versions 13 and 17.

At the moment I don't have time to look at these in detail, but off-the-cuff the following APIs ring a bell to me:

- [allowRendererProcessReuse](https://www.electronjs.org/docs/latest/breaking-changes#removed-appallowrendererprocessreuse)
- [web contents new-content](https://www.electronjs.org/docs/latest/breaking-changes#deprecated-webcontents-new-window-event)
- [nativeWindowOpen](https://www.electronjs.org/docs/latest/breaking-changes#deprecated-nativewindowopen)

If you're able to to perform a cursory triage on if/how the above are used before landing this, that would be great. Either way - we should probably open a separate issue to take a closer look to make sure no unintentional breakages in behavior have been introduced. (Spending a few minutes spinning up the app and playing around with different menu items would also be beneficial.)

**Comment by worldomonation** [comment]
> > Upgraded to 17.0.0 and it finishes the run as expected

However, it seems the issue still persists in the CI environment. I'm trying to pinpoint why this is the case.

[Cancelled run showing some sort of issue persisting](https://teamcity.a8c.com/buildConfiguration/calypso_WpDesktop_DesktopE2ETests/7371773?showRootCauses=false&expandBuildProblemsSection=true&expandBuildTestsSection=true&expandBuildChangesSection=true).

> Could you please create an issue to review the breaking changes that might be caused as a result of this upgrade? In the past, we've been very intentional about upgrading Electron as silent breaking changes have been introduced when upgrading major versions.
> 
> Breaking changes are documented in the [Electron docs](https://www.electronjs.org/docs/latest/breaking-changes), and we'll want to review the changes between major versions 13 and 17.
> 
> At the moment I don't have time to look at these in detail, but off-the-cuff the following APIs ring a bell to me:
> 
>     * [allowRendererProcessReuse](https://www.electronjs.org/docs/latest/breaking-changes#removed-appallowrendererprocessreuse)
> 
>     * [web contents new-content](https://www.electronjs.org/docs/latest/breaking-changes#deprecated-webcontents-new-window-event)
> 
>     * [nativeWindowOpen](https://www.electronjs.org/docs/latest/breaking-changes#deprecated-nativewindowopen)
> 
> 
> If you're able to to perform a cursory triage on if/how the above are used before landing this, that would be great. Either way - we should probably open a separate issue to take a closer look to make sure no unintentional breakages in behavior have been introduced. (Spending a few minutes spinning up the app and playing around with different menu items would also be beneficial.)

I can file an issue. Would it be helpful then, to bypass updating the Desktop app's dependency in the meantime? It would lead to a situation where two versions of Playwright would exist in the dependency tree. This isn't an ideal situation, but from the WPCOM side of things having 1.18 would be quite helpful.

I ran a custom build with the `DEBUG=pw:browser,api` logs enabled and saw that Playwright was generating logs that looked out of place:

```
2022-02-08T20:54:00.882Z pw:browser [pid=795][out] [2022-02-08 20:54:00.881] [desktop:settings] [error] Failed to create settings file:  ENOENT: no such file or directory, open '/calypso/.config/WordPress.com/settings.json'{"errno":-2,"syscall":"open","code":"ENOENT","path":"/calypso/.config/WordPress.com/settings.json"}
12:54:00   2022-02-08T20:54:00.882Z pw:browser [pid=795][out] Error: ENOENT: no such file or directory, open '/calypso/.config/WordPress.com/settings.json'
12:54:00   2022-02-08T20:54:00.882Z pw:browser [pid=795][out]     at Object.openSync (node:fs:585:3)
12:54:00   2022-02-08T20:54:00.882Z pw:browser [pid=795][out]     at Object.func [as openSync] (node:electron/js2c/asar_bundle:5:1800)
12:54:00   2022-02-08T20:54:00.882Z pw:browser [pid=795][out]     at Object.writeFileSync (node:fs:2153:35)
12:54:00   2022-02-08T20:54:00.882Z pw:browser [pid=795][out]     at l (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:34892)
12:54:00   2022-02-08T20:54:00.882Z pw:browser [pid=795][out]     at Object.load (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:35105)
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out]     at a._getAll (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:33824)
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out]     at a.isDebug (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:33899)
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out]     at Object.46784 (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:9054)
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out]     at n (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:964733)
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out]     at e.exports (/home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app/dist/index.js:2:8372)
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out] [2022-02-08 20:54:00.882] [desktop:boot] [info] Booting WordPress.com v7.2.0
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out] [2022-02-08 20:54:00.882] [desktop:boot] [info] App Path: /home/teamcity-1/buildAgent/work/c4a9d5b38c1dacde/desktop/release/linux-unpacked/resources/app
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out] [2022-02-08 20:54:00.883] [desktop:boot] [info] App Data: /calypso/.config/WordPress.com
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out] [2022-02-08 20:54:00.883] [desktop:boot] [info] Settings:{}
12:54:00   2022-02-08T20:54:00.883Z pw:browser [pid=795][out] [2022-02-08 20:54:00.883] [desktop:boot] [info] Proxy: none
12:54:00   2022-02-08T20:54:00.884Z pw:browser [pid=795][out] [2022-02-08 20:54:00.884] [desktop:index] [info] Starting app handlers
12:54:00   2022-02-08T20:54:00.897Z pw:browser [pid=795][out] [2022-02-08 20:54:00.896] [desktop:updater] [info] Updater config:
12:54:00   2022-02-08T20:54:00.897Z pw:browser [pid=795][out] [2022-02-08 20:54:00.897] [desktop:updater] [info] Skipping updater configuration ...
12:54:00   2022-02-08T20:54:00.898Z pw:browser [pid=795][out] [2022-02-08 20:54:00.898] [desktop:index] [info] Waiting for app window to load
12:54:00   2022-02-08T20:54:00.913Z pw:browser [pid=795][err] [795:0208/205400.912865:ERROR:process_singleton_posix.cc(344)] Failed to create /calypso/.config/WordPress.com/SingletonLock: No such file or directory (2)
12:54:00   2022-02-08T20:54:00.913Z pw:browser [pid=795][out] [2022-02-08 20:54:00.913] [desktop:app-instance] [info] App is already running, quitting
12:54:01   2022-02-08T20:54:01.023Z pw:browser [pid=795][err] [795:0208/205401.023033:ERROR:bus.cc(397)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
12:54:01   2022-02-08T20:54:01.023Z pw:browser [pid=795][err] [795:0208/205401.023083:ERROR:bus.cc(397)] Failed to connect to the bus: Failed to connect to socket /var/run/dbus/system_bus_socket: No such file or directory
12:54:01   2022-02-08T20:54:01.024Z pw:browser [pid=795][err] [795:0208/205401.024267:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
12:54:01   2022-02-08T20:54:01.024Z pw:browser [pid=795][err] [795:0208/205401.024300:ERROR:bus.cc(397)] Failed to connect to the bus: Could not parse server address: Unknown address type (examples of valid types are "tcp" and on UNIX "unix")
12:54:01   2022-02-08T20:54:01.064Z pw:browser [pid=795][err]
12:54:01   2022-02-08T20:54:01.064Z pw:browser [pid=795][err] DevTools listening on ws://127.0.0.1:43577/devtools/browser/b095469c-dce5-4c0d-b586-7a9501fc60e1
12:54:01   2022-02-08T20:54:01.064Z pw:browser <ws connecting> ws://127.0.0.1:43577/devtools/browser/b095469c-dce5-4c0d-b586-7a9501fc60e1
12:54:01   2022-02-08T20:54:01.068Z pw:browser <ws connected> ws://127.0.0.1:43577/devtools/browser/b095469c-dce5-4c0d-b586-7a9501fc60e1
12:54:01   2022-02-08T20:54:01.070Z pw:browser [pid=795][err] Waiting for the debugger to disconnect...
12:54:01   2022-02-08T20:54:01.075Z pw:browser [pid=795][err] [847:0208/205401.075771:ERROR:sandbox_linux.cc(377)] InitializeSandbox() called with multiple threads in process gpu-process.
12:55:00   FAIL test/e2e/specs/login.js (60.637 s)
12:55:00     User Can log in
12:55:00       ✕ Log in
```

**Comment by nsakaimbo** [comment]
> However, it seems the issue still persists in the CI environment. I'm trying to pinpoint why this is the case.

Yeah that's odd. Does this fix depend on any environment variables being present locally that aren't in CI?

Believe it or not, `allowRendererProcessReuse` might be the culprit here - for fun you could [try removing it](https://github.com/Automattic/wp-calypso/blob/b73fd66dfa8500c46546a800ded8dc4f66b6b0cf/desktop/app/env.js#L39) and seeing if it helps at all.

In any case, if this seems like a rabbit hole then proceeding with your suggestion of maintaining two Playwright versions in the dependency tree seems reasonable (not ideal, but it's a path forward). However just make sure to revert the Electron version back in the meantime as well. If so, the issue we create should probably be to standardize the Playwright versions beween Calypso and Desktop, and worth mentioning what we discovered so far (i.e. upgrading Electron version seems to address this, but there are discrepancies between local and CI environments). Also that in order to upgrade the Electron version, breaking changes will need to be evaluated/inspected (feel free to copy my notes about breaking changes into the new Github issue).

**Comment by worldomonation** [comment]
> > However, it seems the issue still persists in the CI environment. I'm trying to pinpoint why this is the case.
> 
> Yeah that's odd. Does this fix depend on any environment variables being present locally that aren't in CI?

It should not - I didn't add any variables during my testing locally, and in fact removed several env vars in an effort to reduce the problem surface area. 

> Believe it or not, `allowRendererProcessReuse` might be the culprit here - for fun you could [try removing it](https://github.com/Automattic/wp-calypso/blob/b73fd66dfa8500c46546a800ded8dc4f66b6b0cf/desktop/app/env.js#L39) and seeing if it helps at all.

I tried it both locally and on CI (in a separate branch for testing) and it does not appear to have any impact on the problem in CI.

> In any case, if this seems like a rabbit hole then proceeding with your suggestion of maintaining two Playwright versions in the dependency tree seems reasonable (not ideal, but it's a path forward). However just make sure to revert the Electron version back in the meantime as well. If so, the issue we create should probably be to standardize the Playwright versions beween Calypso and Desktop, and worth mentioning what we discovered so far (i.e. upgrading Electron version seems to address this, but there are discrepancies between local and CI environments). Also that in order to upgrade the Electron version, breaking changes will need to be evaluated/inspected (feel free to copy my notes about breaking changes into the new Github issue).

Certainly, I will proceed with the two different Playwright versions (one for `desktop`, one for everywhere else). Commits reverting the update of `electron` and `playwright` are already pushed to this PR and it looks like everything is in a good state at the moment.

I will file one task/issue for standardizing the Playwright versions and incorporate as much of the logs and findings we've discussed here. I'll then file another child task/issue outlining the need to test the differences between `electron` 12.1 and 17.0.

File: desktop/test/e2e/specs/login.js
Comment (worldomonation) [inline desktop/test/e2e/specs/login.js:122]: This was removed at [suggestion](https://github.com/microsoft/playwright/issues/11920#issuecomment-1032128970) by the Playwright maintainers.

**Review by dpasque — APPROVED** [review]
Wow, this turned into a wild ride! Nice debugging work! 🚀

**Comment by worldomonation** [comment]
@nsakaimbo I've filed the two issues as we agreed: https://github.com/Automattic/wp-calypso/issues/60937 and https://github.com/Automattic/wp-calypso/issues/60938.

**Pull request merged**
